### ISSUE TITLE
Add a `leave space` endpoint

### ISSUE THREAD

**Issue by spaenleh** [graasp/graasp#330]
Users should be able to remove their membership on an item. 
This allows users to "leave" an item. 

Users should only be able to remove their own membership to an item with this endpoint. 
Readers would be allowed to leave a space.

**Comment by dialexo** [comment]
I don't think we need a specific endpoint for that purpose. I think we can reuse the existing delete item membership endpoint, and adjust the control flow such that if `membership.memberId === member.id` then the membership can always be deleted. Otherwise, the normal path applies and we check that the member has a write permission on the item first. Would it make sense?
